toml/src/main/java/com/fasterxml/jackson/dataformat/toml/TomlWriteContext.java

```diff
@@ -0,0 +1,198 @@
+package com.fasterxml.jackson.dataformat.toml;
+
+import com.fasterxml.jackson.core.JsonStreamContext;
+
+final class TomlWriteContext extends JsonStreamContext {
+    /**
+     * Parent context for this context; null for root context.
+     */
+    protected final TomlWriteContext _parent;
+
+    /*
+    /**********************************************************************
+    /* Simple instance reuse slots; speed up things
+    /* a bit (10-15%) for docs with lots of small arrays/objects
+    /**********************************************************************
+     */
+
+    protected TomlWriteContext _child = null;
+
+    /*
+    /**********************************************************************
+    /* Location/state information (minus source reference)
+    /**********************************************************************
+     */
+
+    /**
+     * Value that is being serialized and caused this context to be created;
+     * typically a POJO or container type.
+     */
+    protected Object _currentValue;
+
+    /**
+     * Marker used to indicate that we just received a name, and
+     * now expect a value
+     */
+    protected boolean _gotName;
+
+    /**
+     * Name of the field of which value is to be parsed; only
+     * used for OBJECT contexts
+     */
+    protected String _currentName;
+
+    protected int _basePathLength;
+
+    boolean _inline;
+
+    /*
+    /**********************************************************************
+    /* Life-cycle
+    /**********************************************************************
+     */
+
+    TomlWriteContext(int type, TomlWriteContext parent,
+                                Object currValue, int basePathLength)
+    {
+        super();
+        _type = type;
+        _parent = parent;
+        _basePathLength = basePathLength;
+        _index = -1;
+        _currentValue = currValue;
+        _inline = (parent != null && parent._inline) || type == TYPE_ARRAY;
+    }
+
+    private void reset(int type, Object currValue, int basePathLength) {
+        _type = type;
+        _basePathLength = basePathLength;
+        _currentValue = null;
+        _index = -1;
+        _currentValue = currValue;
+    }
```

Comment:
@yawkat I suspect that `_inline` will need to be recalculated as well? Have not created a test (for a while was considering #258 might be due to this, but that is separate) but thought I'd point this out.

Basically, with reuse, if initially context was Object, reused as Array (or vice versa), this likely would be incorrect.
But before changing would be good to reproduce actual problem. :)

Reply:
Fixed as per #260.